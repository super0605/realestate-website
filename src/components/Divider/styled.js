import styled from "styled-components";

export const DividerContainer = styled.div`
  padding-left: ${({ pleft }) => pleft};
  padding-right: ${({ pright }) => pright};
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  background-color: ${({ bgcolor }) => bgcolor};
`;

export const DividerCotent = styled.div`
  height: ${({ height }) => height};
  width: ${({ width }) => width};
  background-color: ${({ color }) => color};
`;
