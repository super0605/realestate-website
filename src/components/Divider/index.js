import React from "react";
import PropTypes from "prop-types";
import { DividerContainer, DividerCotent } from "./styled";

const Divider = ({ pLeft, pRight, pTop, pBottom, color, bgColor, height, width }) => {
  return (
    <DividerContainer
      pleft={pLeft}
      pright={pRight}
      ptop={pTop}
      pBottom={pBottom}
      bgcolor={bgColor}
      height={height}
      width={width}
    >
      <DividerCotent
        pleft={pLeft}
        pright={pRight}
        ptop={pTop}
        pBottom={pBottom}
        color={color}
        height={height}
        width={width}
      />
    </DividerContainer>
  );
};

Divider.propTypes = {
  pLeft: PropTypes.string,
  pRight: PropTypes.string,
  pTop: PropTypes.string,
  pBottom: PropTypes.string,
  color: PropTypes.string,
  height: PropTypes.string,
  width: PropTypes.string,
  bgColor: PropTypes.string,
};

Divider.defaultProps = {
  pLeft: "0px",
  pRight: "0px",
  pTop: "0px",
  pBottom: "0px",
  color: "#fff",
  height: "1px",
  width: "100%",
};

export default Divider;
