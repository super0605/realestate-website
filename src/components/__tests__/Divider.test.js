import * as React from "react";
import { mount, shallow } from "enzyme";
import Divider from "../Divider";

describe("Divider Component", () => {
  let props;
  let mountedDivider;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedDivider) {
      mountedDivider = bShallow ? shallow(<Divider {...props} />) : mount(<Divider {...props} />);
    }
    return mountedDivider;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
