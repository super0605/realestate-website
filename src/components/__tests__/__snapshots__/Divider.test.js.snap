// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should render without problems 1`] = `
<styled.div
  height="1px"
  pBottom="0px"
  pleft="0px"
  pright="0px"
  ptop="0px"
  width="100%"
>
  <styled.div
    color="#fff"
    height="1px"
    pBottom="0px"
    pleft="0px"
    pright="0px"
    ptop="0px"
    width="100%"
  />
</styled.div>
`;
